aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -407,11 +432,32 @@ module aptos_framework::delegation_pool {
         borrow_global<DelegationPool>(pool_address).observed_lockup_cycle.index
     }
 
+    #[view]
+    /// Return whether the commission percentage for the next lockup cycle is effective.
+    public fun is_next_commission_percentage_effective(pool_address: address): bool acquires NextCommissionPercentage {
+        exists<NextCommissionPercentage>(pool_address) &&
+            timestamp::now_seconds() >= borrow_global<NextCommissionPercentage>(pool_address).effective_after_secs
+    }
+
     #[view]
     /// Return the operator commission percentage set on the delegation pool `pool_address`.
-    public fun operator_commission_percentage(pool_address: address): u64 acquires DelegationPool {
+    public fun operator_commission_percentage(pool_address: address): u64 acquires DelegationPool, NextCommissionPercentage {
         assert_delegation_pool_exists(pool_address);
-        borrow_global<DelegationPool>(pool_address).operator_commission_percentage
+        if (is_next_commission_percentage_effective(pool_address)) {
+            operator_commission_percentage_next_lockup_cycle(pool_address)
+        } else {
+            borrow_global<DelegationPool>(pool_address).operator_commission_percentage
+        }
+    }
+
+    #[view]
+    /// Return the operator commission percentage for the next lockup cycle.
+    public fun operator_commission_percentage_next_lockup_cycle(pool_address: address): u64 acquires DelegationPool, NextCommissionPercentage {
+        if (exists<NextCommissionPercentage>(pool_address)) {
+            borrow_global<NextCommissionPercentage>(pool_address).commission_percentage_next_lockup_cycle
+        } else {
+            operator_commission_percentage(pool_address)
```

Comment:
safer and more explicit to use `borrow_global<DelegationPool>(pool_address).operator_commission_percentage` instead, as this looks like a recursion even it cannot happen due to the branching conditions

Reply:
Done changing it. Good point!